### ISSUE TITLE
False positive for `prefer-reflect` Function.apply

### ISSUE THREAD

**Issue by derekdreery** [eslint/eslint#4318]
Steps to reproduce

```javascript
var x = {
    apply() {}
};
x.apply();
```

Expected: no lint warning
Actual: 
```
Avoid using Function.prototype.apply, instead use Reflect.apply (prefer-reflect) [javascript/eslint]
```

`eslint -v`: `v1.8.0`

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Unfortunately, there's really not much we can do about this. We don't do know what type `x` is, so we can't know for sure that `apply()` is okay to use here. I'd suggest using an inline comment to disable checking of such instances.

**Comment by michaelficarra** [comment]
There's no way to reliably statically distinguish functions from non-functions. What you're asking cannot be done.

**Comment by derekdreery** [comment]
Thanks for the quick response :)
